src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -342,13 +343,31 @@ public static Translation2d[] getModuleTranslations() {
     };
   }
 
-  public Rotation2d getRotationToTranslation(Translation2d translation) {
-
+  public Rotation2d getRotationToTranslation(Pose2d translation) {
     double angle = Math.atan2(translation.getY() - pose.getY(), translation.getX() - pose.getX());
     return Rotation2d.fromRadians(angle);
   }
 
-  public Command pointTowardsTranslation(DoubleSupplier x, DoubleSupplier y) {
+  @AutoLogOutput(key = "Autoaim/Target")
+  public Pose2d getVirtualTarget() {
+
+    Pose2d target = FieldConstants.getSpeaker();
+
+    double distance =
+        Math.sqrt(
+            Math.pow((pose.getX() - target.getX()), 2)
+                + Math.pow((pose.getY() - target.getY()), 2));
+
+    return target.transformBy(
+        new Transform2d(
+                getVelocity().vxMetersPerSecond * AutoAim.shotMap.get(distance).getFlightTime(),
+                getVelocity().vyMetersPerSecond * AutoAim.shotMap.get(distance).getFlightTime(),
+                target.getRotation())
+            .inverse());
+  }
+
+  public Command pointTowardsTranslation(
```

Comment:
maybe this should take a target translation, and have a separate `pointTowardsSpeaker` and `pointTowardsMovingSpeaker` or something along those lines, just to have separation of concerns. Should also probably suffix the method with Cmd to indicate it being a command although we arent super consistent about that elsewhere and im not 100% sold on the practice